source/simulator/assembly.cc

```diff
@@ -1521,7 +1521,7 @@ namespace aspect
       }
     else
       {
-        compute_material_model_input_values (old_solution,
+        compute_material_model_input_values (current_linearization_point,
```

Comment:
I think what @gassmoeller suggests is that the 2 statements in the else-branch are equal to the first 2 statements in the if-branch. So they could be moved before the if and the else-branch completely removed. I think that would make for a nice cleanup.